zproject/backends.py

```diff
@@ -143,7 +144,14 @@ def get_or_create_user(self, username, ldap_user):
         try:
             return get_user_profile_by_email(username), False
         except UserProfile.DoesNotExist:
-            return UserProfile(), False
+            realm = Realm.objects.get(domain=settings.LDAP_APPEND_DOMAIN)
+            full_name = ldap_user.attrs[settings.AUTH_LDAP_USER_ATTR_MAP["full_name"]]
+            short_name = ldap_user.attrs[settings.AUTH_LDAP_USER_ATTR_MAP["full_name"]]
+            if "short_name" in settings.AUTH_LDAP_USER_ATTR_MAP:
+                short_name = settings.AUTH_LDAP_USER_ATTR_MAP["short_name"]
+
+            user_profile = do_create_user(username, None, realm, full_name, short_name)
```

Comment:
Maybe put a comment above this `do_create_user` line to note that LDAP-created users don't have a Zulip password?

Also one related thing to test with this setup is that on the "/#settings" page, the user can fetch their API using their LDAP password;  I'm pretty sure that should work, but we should confirm.